Avoid user_state_populate when there is no change.

infinite-persistence opened the pull request:
## Issue
Every time the app goes into focus and syncs, the GUI updates and does a lot of re-calculations even when the sync hash indicates "no change".

## Changes
At first, I moved `doGetAndPopulated` preferences under the `hasNewData` if-block, since from the function standpoint, it makes sense to not update when there is no new data.

However, I realized the app is stuck in the "still preparing" state at startup, which probably means that it needs to populate at least once.

I found a place that already does that one-off action, but it's for Desktop only. Enabling that does fix the issue, but I think `getWalletSyncPref` is for Desktop only, so I did a simpler version that just updates the prefs.

infinite-persistence commented:
Asking Jessop to review ...

jessopb commented:
This is the way it is because we used to have a wallet before we had a user on desktop. Since you'll likely have a user before you have the updated wallet, you can do things differently. 
It was getting preference before allowing sync to check that preferences allowed sync to happen. You can probably greatly simplify this.
For example,
Line ~400 of app/view.jsx
`      if (IS_WEB) setReadyForSync(true);` will be called as well, and that may be fine?

jessopb commented:
readyForSync and readyForPrefs are two stages for desktop - not for odysee, probably.

infinite-persistence commented:
@jessopb, give it one last glance?

6b97911 to 4c3b057 
- `doHandleSyncComplete`: only call `doGetAndPopulatePreferences` when there is new data.
- But for that to work, we'll need to populate preferences at least once. We'll do that in `doSignIn`.
- We can also remove the "sync/prefs ready" mechanism that was mainly meant for Desktop.

Then came another problem: while trying to spark changes between 2 tabs, `sync/get` was saying "no change" despite the local and server hash being different. I think it is because the `sync_hash + sync/get` combo is both operating on server data, so the hash is the same. I'm guessing this is why we ended up just running `doGetAndPopulatePreferences` every time, since this flag wasn't correct in this scenario.

- Updated `data.changed` to consider both API results and comparison with local hash.

jessopb commented on ui/redux/reducers/tags.js:
```diff
@@ -66,14 +66,6 @@ export default handleActions(
       const { tags } = action.data;
       if (Array.isArray(tags)) {
         const next = tags && tags.filter((tag) => typeof tag === 'string');
```
nice - can delete `tags &&` ?
